src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -863,5 +874,111 @@ std::shared_ptr<coreml::MLModelWrapper> drawing_classifier::export_to_coreml(
   return model_wrapper;
 }
 
+void drawing_classifier::import_from_custom_model(variant_map_type model_data,
+                                               size_t version) {
+  auto model_iter = model_data.find("_model");
+  if (model_iter == model_data.end()) {
+    log_and_throw("The loaded turicreate model must contain '_model'!\n");
+  }
+  const flex_dict& model = variant_get_value<flex_dict>(model_iter->second);
+
+  // For a model trained on integer classes, when saved and loaded back,
+  // the classes are loaded as floats. The following code block casts
+  // the loaded "float" classes back to int.
+  if (model_data.count("classes")) {
+    flex_list classes_list =
+        variant_get_value<flex_list>(model_data.find("classes")->second);
+
+    if (classes_list.size() &&
+        classes_list.begin()->get_type() == flex_type_enum::FLOAT) {
+      flex_list new_classes_list;
+      new_classes_list.reserve(classes_list.size());
+
+      std::transform(classes_list.begin(), classes_list.end(),
+                     std::back_inserter(new_classes_list),
+                     [](flexible_type& ft) { return ft.to<flex_int>(); });
+
+      model_data.erase("classes");
+      model_data.emplace("classes", std::move(new_classes_list));
+
```

Comment:
I think all this could be expressed more concisely as:
```
for (flexible_type& ft : classes_list) {
  ft = ft.to<flex_int>();
}
model_data["classes"] = std::move(classes_list);
```

Reply:
great! I will give that.